src/ariel-os-embassy/src/lib.rs

```diff
@@ -204,7 +182,8 @@ fn init() {
 #[embassy_executor::task]
 #[allow(clippy::too_many_lines)]
 async fn init_task(mut peripherals: hal::OptionalPeripherals) {
-    let spawner = asynch::Spawner::for_current_executor().await;
+    // SAFETY: safe unless intentionally exploited.
```

Comment:
(Might need reflowing)

```suggestion
    // SAFETY: this is executed in a non-public task using only Embassy executors.
```